vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java

```diff
@@ -238,6 +241,30 @@ public String toString()
         });
     }
 
+    /**
+     * Add headers to proxy request that will be used while sending request with given urlPattern
+     * @param headers ExamplesTable representing list of headers with columns "name" and "value" specifying HTTP header
+     * names and values respectively
+     * @param urlPattern The string value of URL-pattern to filter by
+     */
+    @When("I add headers to proxied requests with URL pattern `$urlPattern`:$headers")
+    public void addHeadersToProxyRequest(ExamplesTable headers, String urlPattern)
```

Comment:
`String urlPattern` why not use some flexible mechanism to match URLs like Patter or StringComparisonRule ?